pkg/logs/processor/processor.go

```diff
@@ -52,92 +55,74 @@ func (p *Processor) run() {
-		} else {
-			timestamp := time.Now().UTC().Format(config.DateFormat)
-			extraContent = append(extraContent, []byte(timestamp)...)
-		}
-		extraContent = append(extraContent, ' ')
-
-		// Hostname
-		hostname, err := util.GetHostname()
-		if err != nil {
-			// this scenario is not likely to happen since the agent can not start without a hostname
-			hostname = "unknown"
-		}
-		extraContent = append(extraContent, []byte(hostname)...)
-		extraContent = append(extraContent, ' ')
-
-		// Service
-		service := msg.GetOrigin().LogSource.Config.Service
-		if service != "" {
-			extraContent = append(extraContent, []byte(service)...)
-		} else {
-			extraContent = append(extraContent, '-')
-		}
+func (p *Processor) computeContent(msg message.Message, redactedMessage []byte) ([]byte, error) {
 
-		// Extra
-		extraContent = append(extraContent, []byte(" - - ")...)
+	// TODO Remove occurrences of "severity" (it is now "status")
+	// Compute the status
+	var status string
+	if msg.GetSeverity() != nil {
+		status = string(msg.GetSeverity())
+	} else {
+		status = "info"
+	}
 
-		// Tags
-		extraContent = append(extraContent, msg.GetTagsPayload()...)
-		extraContent = append(extraContent, ' ')
+	// Compute the hostname
+	hostname, err := util.GetHostname()
+	if err != nil {
+		// this scenario is not likely to happen since the agent can not start without a hostname
+		hostname = "unknown"
+	}
 
-		return extraContent
+	// Build the protocol buffer payload
+	payload := &pb.LogPayload{
+		ApiKey: p.apikey,
+		Log: &pb.Log{
+			Message:   string(redactedMessage),
+			Status:    status,
+			Timestamp: msg.GetTimestamp(),
+			Hostname:  hostname,
+			Service:   msg.GetOrigin().LogSource.Config.Service,
+			Source:    msg.GetOrigin().LogSource.Config.Source,
+			Category:  msg.GetOrigin().LogSource.Config.SourceCategory,
+			Tags:      strings.Split(string(msg.GetTagsPayload()), ","),
+		},
 	}
-	return nil
-}
 
-func (p *Processor) computeAPIKeyString(msg message.Message) []byte {
-	sourceLogset := msg.GetOrigin().LogSource.Config.Logset
-	if sourceLogset != "" {
-		return []byte(fmt.Sprintf("%s/%s", p.apikey, sourceLogset))
+	// TODO Remove "logset" from configuration files as it is deprecated
+	// Append logset if necessary
+	logset := msg.GetOrigin().LogSource.Config.Logset
+	if logset != "" {
+		payload.Logset = logset
 	}
-	return p.apikeyString
-}
 
-// buildPayload returns a processed payload from a raw message
-func (p *Processor) buildPayload(apikeyString, redactedMessage, extraContent []byte) []byte {
-	payload := append(apikeyString, ' ')
-	if extraContent != nil {
-		payload = append(payload, extraContent...)
+	// Convert the protocol buffer to a flat byte array
+	body, err := payload.Marshal()
+	if err != nil {
+		return nil, err
 	}
-	payload = append(payload, redactedMessage...)
-	payload = append(payload, '\n')
-	return payload
+
+	// We prepend the body length encoded as a base 128 Varint to the array.
```

Comment:
Can you add an explanation why we do that plz ?

Reply:
also, should that specific logic be in the `sender` instead?

Reply:
I added a quick explanation. Also I feel like the sender should be agnostic to what is being sent, so I preferred having this logic here.